service/tasks/driver.py

```diff
@@ -623,6 +623,8 @@ def get_chain_from_build(
     """
     wait_active_task = wait_for_instance.s(
         instance.id, driverCls, provider, identity, "active")
+    add_security_group = add_security_group_task.si(driverCls, provider, core_identity, instance.id)
+    wait_active_task.link(add_security_group)
```

Comment:
Two things: One, this task is not 'chained', see the modification below. The more important 'thing' is that this task will not work when `has_secret`[1] is False. we need to do one of two things:
1 - if `core_identity` `has_secret` (see https://github.com/cyverse/atmosphere/pull/277/files#diff-545eec308259c382ab4919797a49db52R1344) then:
```
wait_active_task.link(add_security_group)
add_security_group.link(network_start)
start_chain = wait_active_task
return start_chain
```
If has_secret is False:
```
wait_active_task = ...
start_chain = wait_active_task
start_chain.link(network_start)
return start_chain
```

Also, since we're still calling `security_group_init` on instance launch, is this something we could do _before_ launching the instance, rather than after? just checking..

Where security_group_init gets called: https://github.com/xuhang57/atmosphere/blob/1174c918eb159eb116b8d5d28f8d4091b7df571a/service/instance.py#L1711

[1] - We really need a better feature-flag for providers that "have admin accounts" or "do not have admin account and are run by the user".